### ISSUE TITLE
Fix OverrideInIncludedModule counting bug - excludes same module definitions

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#53]
## Context
Code review comment from PR #47: https://github.com/androidbroadcast/Koin-Detekt/pull/47

## Issue
The duplicate type check counts ALL occurrences across the entire file, but doesn't exclude occurrences within the same module being checked. This causes false positives.

### Example False Positive
```kotlin
val module1 = module {
    single<String>(named("a")) { "first" }
}

val module2 = module {
    includes(module1)
    single<String>(named("b")) { "second" }  // Incorrectly flagged as override
    single<String>(named("c")) { "third" }   // Incorrectly flagged as override
}
```

These definitions have different qualifiers and are in the SAME module, so they shouldn't be flagged as overrides. But the current logic (line 138-139) counts all String types in the file.

## Root Cause
```kotlin
val duplicateCount = allTypesInFile.count { it == boundType }
if (duplicateCount > 1) {
    // Reports even if duplicates are in the same module with different qualifiers
}
```

## Proposed Solution
Track which module each type registration belongs to and only count occurrences from DIFFERENT modules:

1. Build a map of `moduleVariable -> List<(boundType, hasQualifier)>`
2. When checking includes, only count types from OTHER modules
3. If counting within the same module, verify there's NO qualifier collision

## Priority
IMPORTANT - Causes false positives on valid code patterns.

## Related
- PR #47: https://github.com/androidbroadcast/Koin-Detekt/pull/47
- Already tracked in issue #50 for general improvements, but this specific bug wasn't called out
